apps/trackers/jira/query.py

```diff
@@ -112,19 +119,40 @@ def generate_description(self):
         """
         self._query["fields"]["description"] = self.description
 
-    # TODO we should not delete other labels
-    # because the engineeting may use them
     def generate_labels(self):
         """
         generate query for Jira labels
         """
-        self._query["fields"]["labels"] = [
-            "SecurityTracking",
-            "Security",
-            f"pscomponent:{self.ps_component}",
-        ] + list(  # add all linked non-empty CVE IDs
-            self.tracker.affects.exclude(flaw__cve_id__isnull=True).values_list(
-                "flaw__cve_id", flat=True
+
+        all_existing_labels = self.tracker.meta_attr.get("labels", [])
+
+        # These labels will be edited here by recreating them as needed.
+        labels_allowed_to_edit_here = set()
+        for lbl in all_existing_labels:
+            if (
+                lbl.startswith("pscomponent:")
+                or lbl in ["SecurityTracking", "Security", "validation-requested"]
+                or CVE_NAME_RE.match(lbl)
+            ):
+                labels_allowed_to_edit_here.add(lbl)
+
+        # These labels are from elsewhere than this method and preserved with their ordering intact.
+        # Because the engineering may use them.
+        labels_to_preserve = [
+            x for x in all_existing_labels if x not in labels_allowed_to_edit_here
+        ]
+        self._query["fields"]["labels"] = labels_to_preserve
+
+        self._query["fields"]["labels"].extend(
+            [
+                "SecurityTracking",
+                "Security",
+                f"pscomponent:{self.ps_component}",
+            ]
+            + list(  # add all linked non-empty CVE IDs
+                self.tracker.affects.exclude(flaw__cve_id__isnull=True).values_list(
+                    "flaw__cve_id", flat=True
+                )
```

Comment:
You can build `self._query["fields"]["labels"]` at once, i.e.:

```
        self._query["fields"]["labels"] = [
            "SecurityTracking",
            ...,
            *list1,
            *list2,
        ]
```

Reply:
I don't understand how that is better or more understandable, but did as you said. (I presume you didn't just mean that _I can_ but that _I should_, otherwise why write it, right?) Please review commit [2c8f57f](https://github.com/RedHatProductSecurity/osidb/pull/428/commits/2c8f57f136f522f3cffc42df037bee5920c849ae)

Reply:
Why not create a variable directly if we can :)? I think this is more compact and understandable. Thank you for the update.